### PR TITLE
stylo: Support :active and :hover quirk

### PR DESCRIPTION
**Pull request by canova**
Reviewed by bholley and emilo on the bugzilla bug.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix [Bug 1355724](https://bugzilla.mozilla.org/show_bug.cgi?id=1355724)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17266)

### CODE REVIEW

File: components/selectors/matching.rs [hunk 1]
```diff
@@ -651,7 +761,12 @@ fn matches_simple_selector<E, F>(
             matches_generic_nth_child(element, 0, 1, true, true, flags_setter)
         }
         Component::Negation(ref negated) => {
-            !negated.iter().all(|ss| matches_simple_selector(ss, element, context, relevant_link, flags_setter))
+            let old_value = context.within_functional_pseudo_class_argument;
+            context.within_functional_pseudo_class_argument = true;
+            let result = !negated.iter().all(|ss| matches_simple_selector(ss, element, context,
```
Comment (emilio) [inline components/selectors/matching.rs]: nit: Please add braces and move the function call to its own line, or indent the arguments (though preferably the first).

File: components/selectors/parser.rs [hunk 2]
```diff
@@ -103,6 +103,11 @@ macro_rules! with_all_bounds {
 
             /// pseudo-elements
             type PseudoElement: $($CommonBounds)* + PseudoElement<Impl = Self>;
+
+            /// Returns whether the selector matches conditions for the :active and
+            /// :hover quirk.
```
Comment (emilio) [inline components/selectors/parser.rs]: Also, this comment is false, this only returns whether the pseudo-class is `:hover` or `:after`, right?

File: components/style/gecko/wrapper.rs [hunk 3]
```diff
@@ -1477,12 +1474,21 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
             NonTSPseudoClass::MozMeterSubSubOptimum |
             NonTSPseudoClass::MozAutofill |
             NonTSPseudoClass::MozAutofillPreview => {
-                // NB: It's important to use `intersect` instead of `contains`
-                // here, to handle `:any-link` correctly.
+                /// FIXME: This can/should probably be contains() now that any-link
+                // (which depends in multiple bits) is handled in its own case below.
```
Comment (emilio) [inline components/style/gecko/wrapper.rs]: Intersect is fine, you can remove or revert the comment.

File: components/selectors/parser.rs (hunk 2 above)
Comment (canaltinova) [inline components/selectors/parser.rs]: Ah, this method changed so many times and I forgot to change the comment at some point :)

**Pull request merged**